addons/project/models/project_project.py

```diff
@@ -862,12 +862,15 @@ def _create_analytic_account_from_values(self, values):
                     'name': _('Default'),
                     'company_id': False,
                 })
+
+        # When creating new projects, we do not want the analytic account's partner to be set
+        # as it could cause unwanted company validation errors
         analytic_account = self.env['account.analytic.account'].create({
             'name': values.get('name', _('Unknown Analytic Account')),
             'company_id': company.id,
-            'partner_id': values.get('partner_id'),
```

Comment:
Why not setting the company of the partner to the project instead of completely removing the partner on the AA? 🤔

Reply:
Actually, an old version of this PR did just that, but we thought it was a big change to the business logic for stable to add a completely new constraint that projects must have the same company as the associated partner if any. Especially since it slightly goes against the goal of [this commit](https://github.com/odoo/odoo/commit/16a07ed1bfd33a193b363f3e87aab3f9bd6f8bd0)  
I think it also breaks a few tests which we'd have to modify, but so does this version so...  
I can make it do that again if you think that's best

Reply:
If the partner and AA should have the same company than we should set the company instead of completely removing the partner on the AA generated to solve the issue. 😕 

Because functionally, that partner could be really useful, we don't really know. 🤷🏼‍♂️